Password reset link decay bug

jlikhuva opened the pull request:
@caesar2164 @caseylitton @stvstnfrd  PR for password reset link 'decay' bug.

Fixes the issue where no error is issued if, when attempting to reset their password, the user
repeats the new password incorrectly.  The code change ensures that the user receives an error message and that the reset link can still be used after the unsuccessful reset attempt.

See this [Trello Card](https://trello.com/c/XHaarPfj/872-entering-a-different-password-in-the-two-password-reset-fields-invalidates-the-password-reset-link) for further details.

@jspayd @potsui

caesar2164 commented on common/lib/xmodule/xmodule/capa_base.py:
```diff
@@ -137,7 +138,15 @@ class CapaFields(object):
+        help=_("Number of times the student must attempt answering the question before the Show Answer button appears."),
+        values={"min": 0},
+        default=0,
+        scope=Scope.settings
```
this shouldn't be here I don't think...this is from the "Show answer after # attempts" PR.

caesar2164 commented on common/lib/xmodule/xmodule/capa_base.py:
```diff
@@ -899,9 +908,15 @@ def answer_available(self):
+            required_attempts = self.attempts_before_showanswer_button
+            if self.max_attempts and required_attempts >= self.max_attempts:
+                required_attempts = self.max_attempts
+            return self.attempts >= required_attempts
```
this addition shouldn't be in this PR.

caesar2164 commented on common/lib/xmodule/xmodule/capa_base.py:
```diff
@@ -899,9 +908,15 @@ def answer_available(self):
+            return self.attempts >= required_attempts
         elif self.showanswer == SHOWANSWER.ALWAYS:
             return True
-
```
same with this

caesar2164 commented on common/lib/xmodule/xmodule/capa_base_constants.py:
```diff
@@ -16,6 +16,7 @@ class SHOWANSWER(object):
     CORRECT_OR_PAST_DUE = "correct_or_past_due"
     PAST_DUE = "past_due"
     NEVER = "never"
+    AFTER_ATTEMPTS = "after_attempts"
```
remove

caesar2164 commented on common/lib/xmodule/xmodule/tests/test_capa_module.py:
```diff
@@ -361,6 +361,65 @@ def test_showanswer_past_due(self):
                                             graceperiod=self.two_day_delta_str)
         self.assertFalse(still_in_grace.answer_available())
 
+    def test_showanswer_after_attempts(self):
```
changes in this file also shouldn't be here

stvstnfrd commented:
@jlikhuva As @caesar2164 mentions, this looks like it contains unrelated changes.

I suspect that when you created this branch, you accidentally based it off of your other feature branch, instead of `master`.

We'll want to rebase this branch to remove the extraneous commits;
let me know if you want a hand with this.

jlikhuva commented:
@caesar2164 , so I rebased my branch, deleting the commits not related to this feature. I then pushed (--force). That results in conflicts below. I already fixed one of them (the one in Authors file)
I'm not familiar with the code resulting in the other conflict, so I haven't touched that.
@stvstnfrd

potsui commented:
@jlikhuva it looks like the `forms.py` conflicts are changes @caesar2164 made in an old commit that's already been merged, and you also have in your branch: ac405c829e11c233fc03f5ec0f77b0da7978cd73

caesar2164 commented:
first two commits of this don't seem necessary...

stvstnfrd commented:
I took a peek upstream and it looks like edX has already fixed this upstream [1].
This code is also a part of the ongoing merge.

We probably want to either try cherry-picking this commit or waiting until the merge,
instead of writing a custom fix.

Thoughts @caesar2164 @caseylitton ?

[1] https://github.com/edx/edx-platform/commit/3f21b0718fb25303f2a810410bd69c8e77d06bc9

caesar2164 commented:
hmmmm, not sure.

caesar2164 commented:
@jlikhuva, could you try doing a cherry pick (in a new branch of of our master) of the commit that @stvstnfrd linked above?

stvstnfrd commented:
@jlikhuva Can we close this now that we merged the other version?
